src/main/java/com/hms_networks/sc/cumulocity/data/CConnectorChildUpdate.java

```diff
@@ -0,0 +1,270 @@
+
+  /** JSON key for firmware version. */
+  private static final String C8Y_VERSION_KEY = "version";
+
+  /** JSON key for firmware URL. */
+  private static final String C8Y_URL_KEY = "url";
+
+  /** JSON key for hardware model. */
+  private static final String C8Y_MODEL_KEY = "model";
+
+  /** JSON key for hardware revision. */
+  private static final String C8Y_REVISION_KEY = "revision";
+
+  /** JSON key for hardware serial number . */
+  private static final String C8Y_SERIAL_KEY = "serialNumber";
+
+  /** Cumulocity ID for the child - derived from from tag names */
+  private String childDeviceId;
+
+  /** Value of the HardwareModel tag. */
+  private String hardwareModel = null;
+
+  /** Value of the HardwareVersion tag - used for c8y_Hardware revision key. */
+  private String hardwareRevision = null;
+
+  /** Value of the HardwareSerialNumber tag. */
+  private String hardwareSerialNum = null;
+
+  /** Value of the FirmwareName tag. */
+  private String firmwareName = null;
+
+  /** Value of the FirmwareVersion tag. */
+  private String firmwareVersion = null;
+
+  /** Value of the FirmwareUrl tag. */
+  private String firmwareUrl = null;
+
+  /** Track the number of tags read. */
+  private int numReadTags = 0;
+
+  /**
+   * Constructor for the CConnectorChildUpdate class. This constructor will read the tags for the
+   * child ID provided.
+   *
+   * @param childId the id of the child device
+   * @since 1.0.0
+   */
+  public CConnectorChildUpdate(String childId) {
+    this.childDeviceId = childId;
+    readHardwareTags();
+    readFirmwareTags();
+  }
+
+  /**
+   * Try to read the hardware specific c8y tags.
+   *
+   * @since 1.0.0
+   */
+  private void readHardwareTags() {
+    hardwareModel = tryGetMetaDataTag(TAG_POSTFIX_HARDWARE_MODEL);
+    hardwareRevision = tryGetMetaDataTag(TAG_POSTFIX_HARDWARE_REVISION);
+    hardwareSerialNum = tryGetMetaDataTag(TAG_POSTFIX_HARDWARE_SERIAL_NUM);
+  }
+
+  /**
+   * Try to read the firmware specific c8y tags.
+   *
+   * @since 1.0.0
+   */
+  private void readFirmwareTags() {
+    firmwareName = tryGetMetaDataTag(TAG_POSTFIX_FIRMWARE_NAME);
+    firmwareVersion = tryGetMetaDataTag(TAG_POSTFIX_FIRMWARE_VERSION);
+    firmwareUrl = tryGetMetaDataTag(TAG_POSTFIX_FIRMWARE_URL);
+  }
+
+  /**
+   * Try to read the tag value for the given postfix. If the tag does exist, then return the value
+   * as a string.
+   *
+   * @param postfix the specific tag ending to read
+   * @return null, if tag could not be read. Otherwise, the value of the tag as a string.
+   * @since 1.0.0
+   */
+  private String tryGetMetaDataTag(String postfix) {
+    try {
+      TagControl tc = new TagControl(childDeviceId + "/" + TAG_METADATA_TOKEN + "/" + postfix);
+      numReadTags++;
+      return tc.getTagValueAsString();
+      // if the tag does not exist, it is not an EWException that is thrown as JavaDocs suggests, so
+      // here we catch all
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL(
```

Comment:
Is this something that needs to be logged at the critical log level? Especially when the message starts with "warning," I think it may be more appropriate at the warning log level (or similar)

Reply:
will change to warning